lib/TelemetryFeed.jsm

```diff
@@ -63,10 +69,73 @@ this.TelemetryFeed = class TelemetryFeed {
     this._onEventsTelemetryPrefChange = this._onEventsTelemetryPrefChange.bind(this);
     this._prefs.observe(EVENTS_TELEMETRY_PREF, this._onEventsTelemetryPrefChange);
     this._classifySite = classifySite;
+    this._addWindowListeners = this._addWindowListeners.bind(this);
+    this.handleEvent = this.handleEvent.bind(this);
   }
 
   init() {
     Services.obs.addObserver(this.browserOpenNewtabStart, "browser-open-newtab-start");
+    // Add pin tab event listeners on future windows
+    Services.obs.addObserver(this._addWindowListeners, DOMWINDOW_OPENED_TOPIC);
+    // Listen for pin tab events on all open windows
+    for (let win of Services.wm.getEnumerator("navigator:browser")) {
+      if (win.closed || PrivateBrowsingUtils.isWindowPrivate(win)) {
```

Comment:
Hmm, this ping was still sent when I pinned tabs in the private window.

Reply:
It's because I was checking for private windows only on currently open windows and then

```
Services.obs.addObserver(this._addWindowListeners, DOMWINDOW_OPENED_TOPIC);
```
I was adding PinTab event listeners for all new windows (private or not).